### PR TITLE
sbt 1.5.0

### PR DESCRIPTION
**Pull request by vincenzobaz**
This PR upgrades the project to sbt 1.5.0 both in the build system and in the sbt infrastructure used to run snippets.

### CODE REVIEW

File: sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/RuntimeErrorLogger.scala [hunk 1]
```diff
@@ -79,7 +79,6 @@ object RuntimeErrorLogger {
     showSuccess := false,
     logManager := sbt.internal.LogManager.withLoggers(
       (task, state) => defaultScreen(ConsoleOut.printWriterOut(NoOp()), suppressedMessage(task, state)),
-      relay = _ => clientLogger
```
Comment (vincenzobaz) [inline sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/RuntimeErrorLogger.scala:82]: This modification is the only one that I am worried about as it potentially changes the output

File: sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/SbtScastiePlugin.scala [hunk 2]
```diff
@@ -20,7 +20,8 @@ object SbtScastiePlugin extends AutoPlugin {
           val r = compilers.value
           //compile bridge to init everything on reload
           r.scalac() match {
-            case c: AnalyzingCompiler => c.force(streams.value.log)
+            case c: AnalyzingCompiler =>
+              c.provider.fetchCompiledBridge(c.scalaInstance , streams.value.log)
```
Comment (vincenzobaz) [inline sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/SbtScastiePlugin.scala:24]: `force` has been removed, but its [building blocks](https://github.com/sbt/zinc/blob/1e0f5df990c6423f9dec875d7166fa9451b36716/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/AnalyzingCompiler.scala#L238) are still available

File: sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/RuntimeErrorLogger.scala (hunk 1 above)
Comment (vincenzobaz) [reply sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/RuntimeErrorLogger.scala:82]: also `extraLoggers` was deprecated

Comment (OlegYch) [reply sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/RuntimeErrorLogger.scala:82]: i don't think extraloggers was doing anything, but we certainly need the relay

Comment (vincenzobaz) [reply sbt-scastie/src/main/scala/com.olegych.scastie.sbtscastie/RuntimeErrorLogger.scala:82]: I can easily use an  `Appender` from `sbt.internal` which formats messages exactly like the log4j one in the method which receives `Throwable`. I will try later.

I am worried that the code will be invoked only with Throwable and not on all `LogEvent`s

**Comment by OlegYch** [comment]
@vincenzobaz i did some more cleanup in sbt15 branch
"capture runtime errors" in SbtActorTest will fail without correct logger
though currently sbt seems to hang, i'm not sure yet what's the reason

**Comment by vincenzobaz** [comment]
Did you manage to get it working?

**Comment by OlegYch** [comment]
> Did you manage to get it working?

yep, it's deployed now
thank you for the help

**Comment by vincenzobaz** [comment]
What was wrong? Locally I still have nonterminating tests and travis does not seem to be happy/

**Comment by OlegYch** [comment]
@vincenzobaz i did some further fixing https://github.com/scalacenter/scastie/commit/7994408f9ceb867f9d461a79dac70760de2d9a32 https://github.com/scalacenter/scastie/commit/8d4aef02a65971b4815e39b19b500216ebcc0a33

**Comment by vincenzobaz** [comment]
Cool, I did not see what was happening on master. Thank you @OlegYch, I am happy that this is merged !

**Pull request merged**